lib/mongoProvider.js

```diff
@@ -103,7 +103,7 @@ MongoCollection.prototype.find = function (query, req) {
     return self.db().then(function (db) {
         return self.beforeFindListeners.fire(query, req).then(function () {
 
-            return q.ninvoke(db.collection(self.name).find(query), "toArray").then(function(res) {
+            return q.ninvoke(db.collection(self.name).find(query, req), "toArray").then(function(res) {
```

Comment:
You are passing here the `req` into the mongo find function. I don't think it is interested in it. Am I missing something?